smelter-core/src/pipeline/hls/hls_input.rs

```diff
@@ -133,8 +145,15 @@ impl HlsInput {
         video_decoders: HlsInputVideoDecoders,
         buffer: InputBuffer,
     ) -> Result<(Track, Receiver<PipelineEvent<Frame>>), InputInitError> {
+        let stats_sender =
+            HlsInputTrackStatsSender::new(input_ref, &ctx.stats_sender, HlsTrack::Video);
         let (frame_sender, frame_receiver) = bounded(5);
-        let state = StreamState::new(ctx.queue_sync_point, stream.time_base(), buffer);
+        let state = StreamState::new(
+            ctx.queue_sync_point,
+            stream.time_base(),
+            buffer,
+            stats_sender,
+        );
```

Comment:
```suggestion
        let state = StreamState::new(ctx, stream.time_base(), buffer, HlsTrack::Video);
```
and create sender inside new

I would also recommend renaming HlsTrack to e.g. TrackKind. Hls is not necessary in private type and without Kind is unclear whethre it represents Track state or Track type